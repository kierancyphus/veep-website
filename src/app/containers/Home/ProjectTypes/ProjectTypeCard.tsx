import React, { FC, ReactNode } from 'react'
import {
  Card,
  CardContent,
  CardHeader,
  Typography,
  Box,
  CardActionArea,
  CardMedia,
  Button,
  Link,
} from '@material-ui/core'
import { WithProjectType, WithProjectTypeSetter } from 'types/projects'
import { Link as RouterLink } from 'react-router-dom'

export interface ProjectTypeCardProps
  extends WithProjectType,
    WithProjectTypeSetter {
  imageSrc?: string
  description: string
  Icon?: ReactNode
  title: string
  links: Record<string, string>
}

export const getLink = (projectType: number, text: string): string =>
  `/projects?projectType=${projectType}&time=${text
    .split(' ')[0]
    .toLowerCase()}`

export const ProjectTypeCard: FC<ProjectTypeCardProps> = ({
  projectType,
  description,
  Icon,
  setProjectType,
  imageSrc,
  title,
  links,
}) => (
  <Card elevation={5}>
    <Box padding={3}>
      <CardActionArea onClick={() => setProjectType(projectType)}>
        {Icon && Icon}
        {imageSrc && <CardMedia src={imageSrc} />}
        <CardHeader title={title} titleTypographyProps={{ align: 'center' }} />
        <CardContent>
          <Typography variant="body1" align="center">
            {description}
          </Typography>
        </CardContent>
      </CardActionArea>
      <Box display="flex" justifyContent="center" width="100%">
        <Link component={RouterLink} to="/projects" style={{ width: '100%' }}>
          <Button fullWidth>View Projects</Button>
        </Link>
      </Box>
    </Box>
  </Card>
)
