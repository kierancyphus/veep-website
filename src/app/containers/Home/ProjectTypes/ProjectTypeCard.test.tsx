import React from 'react'
import { ProjectTypeCard, ProjectTypeCardProps } from './ProjectTypeCard'
import { render, fireEvent } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'

describe('ProjectTypeCard', () => {
  const props: ProjectTypeCardProps = {
    setProjectType: jest.fn(),
    projectType: 0,
    description: 'description',
    title: 'title',
    links: {
      button: '/path',
    },
  }

  it('renders without errors', () => {
    const { container: firstChild } = render(<ProjectTypeCard {...props} />, {
      wrapper: BrowserRouter,
    })

    expect(firstChild).toMatchSnapshot()
  })

  it('calls setValue with the right value on click', () => {
    const { getByText } = render(<ProjectTypeCard {...props} />, {
      wrapper: BrowserRouter,
    })

    fireEvent.click(getByText(props.description))

    expect(props.setProjectType).toHaveBeenCalledWith(0)
  })
})
