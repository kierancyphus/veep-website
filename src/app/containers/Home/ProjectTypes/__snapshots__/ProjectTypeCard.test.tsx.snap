// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectTypeCard renders without errors 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation5 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-alignCenter MuiTypography-displayBlock"
            >
              title
            </span>
          </div>
        </div>
        <div
          class="MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
          >
            description
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiBox-root MuiBox-root-4"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="/projects?projectType=0&time=button"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              button
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
